### ISSUE TITLE
[java] UnnecessaryBoxing false positive when calling `List.remove(int)`

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#5948]
**Affects PMD Version:** 7.16.0

**Rule:** <https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#unnecessaryboxing>

**Description:** The following code triggers UnnecessaryBoxing. If I remove the call to .intValue(), the wrong overload of ArrayList.remove() is called.

**Code Sample demonstrating the issue:**

```java
        List<String> l = new ArrayList<>();
        Integer i = 3;

        l.remove(i.intValue());
```

**Expected outcome:**

No violation

**Running PMD through:** *IntelliJ-Plugin*
